code/modules/surgery/organs/organ_internal.dm

```diff
@@ -199,21 +200,29 @@ INITIALIZE_IMMEDIATE(/obj/item/organ)
 	var/delta = damage - prev_damage
 	if(delta > 0)
 		if(damage >= maxHealth)
-			organ_flags |= ORGAN_FAILING
 			return now_failing
 		if(damage > high_threshold && prev_damage <= high_threshold)
 			return high_threshold_passed
 		if(damage > low_threshold && prev_damage <= low_threshold)
 			return low_threshold_passed
 	else
-		organ_flags &= ~ORGAN_FAILING
 		if(prev_damage > low_threshold && damage <= low_threshold)
 			return low_threshold_cleared
 		if(prev_damage > high_threshold && damage <= high_threshold)
 			return high_threshold_cleared
 		if(prev_damage == maxHealth)
 			return now_fixed
 
+/**check_failing_thresholds
+ * input: mob/organ_owner (a mob, the owner of the organ we call the proc on)
+ * output: failing organ flags on a failing/no longer failing organ.
+ */
+/obj/item/organ/proc/check_failing_thresholds(mob/organ_owner)
+	if(!(organ_flags & ORGAN_FAILING) && damage >= maxHealth)
+		organ_flags |= ORGAN_FAILING
+	if((organ_flags & ORGAN_FAILING) && (damage < maxHealth))
+		organ_flags &= ~ORGAN_FAILING
```

Comment:
you can probably remove the organ_flags & ORGAN_FAILING check

Reply:
I was hesitant on it because constantly calling for the organ flag being added/removed felt like it was just spamming for no reason, but I'll change it.

Reply:
It's a trivial cost.